timemachine/cpp/src/nonbonded.cu

```diff
@@ -292,64 +295,61 @@ void Nonbonded<RealType, Interpolated>::execute_device(
 
     // identify which tiles contain interpolated parameters
 
-    const int B = (N+32-1)/32;
     const int tpb = 32;
+    const int B = (N+tpb-1)/tpb;
 
     dim3 dimGrid(B, 3, 1);
-
-    // (ytz) see if we need to rebuild the neighborlist.
-    // (ytz + jfass): note that this logic needs to change if we use NPT later on since a resize in the box
-    // can introduce new interactions.
-    k_check_rebuild_coords_and_box<RealType><<<B, tpb, 0, stream>>>(
-        N,
-        d_x,
-        d_nblist_x_,
-        d_box,
-        d_nblist_box_,
-        nblist_padding_,
-        d_rebuild_nblist_
-    );
-    gpuErrchk(cudaPeekAtLastError());
-
-    // we can optimize this away by doing the check on the GPU directly.
-    gpuErrchk(cudaMemcpyAsync(p_rebuild_nblist_, d_rebuild_nblist_, 1*sizeof(*p_rebuild_nblist_), cudaMemcpyDeviceToHost, stream));
-    gpuErrchk(cudaStreamSynchronize(stream)); // slow!
-
-    if(p_rebuild_nblist_[0] > 0) {
-
-        // (ytz): update the permutation index before building neighborlist, as the neighborlist is tied
-        // to a particular sort order
+    // If we have to sort, we also have to rebuild the neighborlist
+    if (sort_indices) {
         if(!disable_hilbert_) {
             this->hilbert_sort(d_x, d_box, stream);
         } else {
-            k_arange<<<B, 32, 0, stream>>>(N, d_perm_);
+            k_arange<<<B, tpb, 0, stream>>>(N, d_perm_);
             gpuErrchk(cudaPeekAtLastError());
         }
-
-        // compute new coordinates, new lambda_idxs, new_plane_idxs
-        k_permute<<<dimGrid, tpb, 0, stream>>>(N, d_perm_, d_x, d_sorted_x_);
-        gpuErrchk(cudaPeekAtLastError());
-        nblist_.build_nblist_device(
+    } else {
+        // (ytz) see if we need to rebuild the neighborlist.
+        // (ytz + jfass): note that this logic needs to change if we use NPT later on since a resize in the box
```

Comment:
we can update this comment with: this logic for NPT is correct, but can probably be made more efficient.